### DISCUSSION TITLE
Unable to deploy Typesense dashboard on K8s cluster

### DISCUSSION THREAD

**Discussion by wali97** [bfritscher/typesense-dashboard#70] [General]
I have self-hosted a 3node typesense ha cluster on my k8s,i'm trying to deploy dashboard using the latest docker image mentioned in this registry...i keep getting network error when i try to access the dashboard...attaching my deployment manifest file which i'm using to deploy the dashboard.
Any assistance would be highly appreciated,thanks.

[type-dashboard.txt](https://github.com/user-attachments/files/16778766/type-dashboard.txt)

**Comment by bfritscher** [comment]
Hi, this app is frontend only (your browser makes the requests to the api), I suppose your 
- name: TYPESENSE_SERVER_URL
          value: "http://<servicename.namespacename.svc.cluster.local:8108"
is not reachable from your browser.

**Reply by wali97** [reply]
yes,this is not publicly hosted...this typesense server url will be accessible from within the cluster...but i need to host the dashboard publicly...i deployed the dashboard in same namespace as of typesense server....
I want my typesense dashboard to requests from browser and then route that request to typesense server internally and get response..
Is this achievable on kubernetes for this tool?

**Reply by bfritscher** [reply]
As mentioned, a connection is required between your browser's JavaScript and the TypeScript API (we use the default JavaScript-Typesense API package). Writing a custom API just to proxy it on the server side doesn’t make much sense for an admin tool.

You have two options: either you expose the API to the public so your browser can access it directly, or you use Kubernetes CLI tools to proxy your dashboard service. In the latter case, you would use `localhost:proxyport` as the `server_url`. With this approach, you wouldn’t need the webapp version—you could simply download the Electron package from the release page, but you are required to recreate the proxy each time you use the app on your machine.
